test/e2e/ownerrefs_finalizers_test.go

```diff
@@ -44,12 +49,20 @@ import (
 var _ = Describe("Ensure OwnerReferences and Finalizers are resilient with FailureDomains and ClusterIdentity", func() {
 	const specName = "owner-reference"
 	Setup(specName, func(testSpecificSettingsGetter func() testSettings) {
-		// Before running the test create the secret used by the VSphereClusterIdentity to connect to the vCenter.
-		BeforeEach(func() {
-			createVsphereIdentitySecret(ctx, bootstrapClusterProxy)
-		})
-
 		capi_e2e.QuickStartSpec(ctx, func() capi_e2e.QuickStartSpecInput {
+			input := testSpecificSettingsGetter()
+			username, ok := input.Variables["VSPHERE_USERNAME"]
+			if !ok {
+				username = os.Getenv("VSPHERE_USERNAME")
```

Comment:
Would it make sense to use e2eConfig.GetVariable()? (which checks env vars & e2e config)

Reply:
No.
We want to use test-specific variables computed during setup, not the variables existing in the e2eConfig
 struct that was loaded when running the SynchronizedBeforeSuite (this is required to run this test with vcsim).

Reply:
I got the part about why we use input.Variables. I'm not sure how falling back to environment variables makes sense to be honest. But it's okay

Reply:
Ah. Are we using the one from the environment for the non-vcsim tests? Would it make sense to add this as a comment as well?

Reply:
When running on CI, VSPHERE_USERNAME and VSPHERE_PASSWORD are provided as env variables; on vcsim they are provided as a test specific variables.
With the fall back we can handle both scenarios

Reply:
Got it

Reply:
Godoc comment would be nice to clarify this point for future us